pmlc/rt/jit_executable.h

```diff
@@ -15,6 +15,7 @@ namespace pmlc::rt {
 std::unique_ptr<Executable>
 makeJitExecutable(const std::shared_ptr<pmlc::compiler::Program> &program,
                   std::shared_ptr<Device> device,
+                  llvm::ArrayRef<void *> preParams,
```

Comment:
I would have thought that the `Device` we are passing in here would be sufficient? Can we pass the `device*` to the jitted program?

Reply:
The thinking was that the JIT logic should not know or care about any extra parameters that're needed by a target.  But I'm no longer convinced this is necessary at all, if we're going to a more opaque (from the PoV of the caller) model.